module-2/backend/week_one.md

```diff
@@ -1,32 +1,69 @@
+
 2. What is Sinatra?
+  Sinatra is a domain specific language used with Ruby in order to create web based applications. In a nutshell, it's a series of pre-written methods that can be included in an application that can turn it into a ruby web app.
 4. What is MVC?
+  MVC stands for Models Views and Controllers. MVC is a representation of the flow of data from the client to the database. The Controller stores the routes and facilitates the client request and the route it needs to take to retrieve or modify that request. The Model contains the data and logic. And the views houses the code for visual representation.
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+  We follow convention when creating our action/path names in order to make it easier for other developers to understand and maintain our code.
 6. What types of variables are accessible in our view templates without explicitly passing them?
+  Instance variables
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
-  
+
+  @count = 1
   ```ruby
   get '/horses' do
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+```ruby
+get '/horses' do
+  name = "Mr. Ed"
+  erb :index
+end
+```
 9. What's the purpose of ERB?
+  The purpose of ERB is to provide a flexible solution to render HTML pages with embedded dynamic ruby content  
 10. Why do I need a development AND test database?
+  You need to keep them separate so you can isolate and control your test cases for errors.
 11. What is CRUD and why is it important?
-12. What does HTTP stand for? 
+  CRUD stands for Create Read Update Delete. It's important because it describes the functions for interacting with a database.
+12. What does HTTP stand for?
+  Hyper Text Transfer Protocol
 13. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+  There are two ways to interpolate ruby. One is : <% %>. The second is : <%=  %>. The difference is the one with the equal sign will output the value or end result to the display while the other will not return the value to display.
 14. What's an ORM?
+  Object Relational Mapper
 15. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+  ActiveRecord
 16. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-17. What's a migration? 
+  1) get read index page
+  2) get show index page
+  3) get create new form
+  4) post submit form
+  5) put/patch update resource by edit form
+  6) delete destroys resource
+  7) get read edit form 
+
+17. What's a migration?
+  Migration is the creation of a database table structure

```

Comment:
I would say that a migration holds the **instructions** to create the database structure.